src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -470,6 +492,15 @@ static inline void uct_dc_mlx5_iface_dci_alloc(uct_dc_mlx5_iface_t *iface, uct_d
     ucs_assert(uct_dc_mlx5_ep_from_dci(iface, ep->dci) == NULL);
     iface->tx.dcis[ep->dci].ep = ep;
     pool->stack_top++;
+    if (ep->flags & UCT_DC_MLX5_EP_FLAG_INVALID) {
+        uct_ib_mlx5_md_t *md = ucs_derived_of(iface, uct_ib_mlx5_md_t);
+        ucs_status_t status;
+        UCT_DC_MLX5_TXQP_DECL(txqp, txwq);
+        UCT_DC_MLX5_IFACE_TXQP_GET(iface, ep, txqp, txwq);
+
+        status = uct_ib_mlx5_modify_qp_state(md, &txwq->super, IBV_QPS_ERR);
+        ucs_assert_always(status == UCS_OK);
```

Comment:
better put it in a separate function in C file. could share code with uct_dc_mlx5_ep_invalidate, and also add a ucs_debug print.